File: spring/src/main/java/dev/x402/spring/X402Config.java
package dev.x402.spring;

import java.util.function.Function;

/**
 * Resolved server-side configuration.
 *
 * <p>Kept identical in spirit to the other SDK references so all default to
 * the same platform.
 */
public final class X402Config {

    /**
     * Default platform base URL — identical across all x402 SDKs.
     * TODO: replace with the production platform domain before GA.
     */
    public static final String DEFAULT_BASE_URL = "https://api.x402.dev";

    public final String keyId;
    public final String secret;
    public final String baseUrl;
    /** "production" | "sandbox" */
    public final String env;

    public X402Config(String keyId, String secret, String baseUrl, String env) {
        this.keyId = keyId;
        this.secret = secret;
        this.baseUrl = baseUrl;
        this.env = env;
    }

    /**
     * Resolves config via the provided getter (pass {@code System::getenv} in
     * production; an injected function in tests).
     *
     * @throws IllegalStateException if X402_API_KEY or X402_SECRET is missing
     */
    public static X402Config resolve(Function<String, String> getenv) {
        Function<String, String> g = (getenv != null) ? getenv : System::getenv;

        String keyId = nullToEmpty(g.apply("X402_API_KEY"));
        String secret = nullToEmpty(g.apply("X402_SECRET"));
        if (keyId.isEmpty() || secret.isEmpty()) {
            throw new IllegalStateException(
                    "x402: X402_API_KEY and X402_SECRET must be set (server-side env)");
        }

        String mode = nullToEmpty(g.apply("X402_ENV")).toLowerCase(java.util.Locale.ROOT);
        if (mode.isEmpty()) {
            mode = "production";
        }

        String base = nullToEmpty(g.apply("X402_BASE_URL"));
        if (base.isEmpty()) {
            base = DEFAULT_BASE_URL;
        }
        base = stripTrailingSlash(base);

        String env = "sandbox".equals(mode) ? "sandbox" : "production";
        return new X402Config(keyId, secret, base, env);
    }

    private static String nullToEmpty(String s) {
        return (s == null) ? "" : s;
    }

    private static String stripTrailingSlash(String s) {
        int end = s.length();
        while (end > 0 && s.charAt(end - 1) == '/') {
            end--;
        }
        return s.substring(0, end);
    }
}

File: aspnetcore/src/X402Config.cs
using System;

namespace X402.AspNetCore;

/// <summary>
/// Resolved server-side configuration.
///
/// <para>Kept identical in spirit to the other SDK references so all default
/// to the same platform.</para>
/// </summary>
public sealed class X402Config
{
    /// <summary>
    /// Default platform base URL — identical across all x402 SDKs.
    /// TODO: replace with the production platform domain before GA.
    /// </summary>
    public const string DefaultBaseUrl = "https://api.x402.dev";

    public string KeyId { get; }
    public string Secret { get; }
    public string BaseUrl { get; }

    /// <summary>"production" | "sandbox"</summary>
    public string Env { get; }

    public X402Config(string keyId, string secret, string baseUrl, string env)
    {
        KeyId = keyId;
        Secret = secret;
        BaseUrl = baseUrl;
        Env = env;
    }

    /// <summary>
    /// Resolves config via the provided getter (pass <c>Environment.GetEnvironmentVariable</c>
    /// in production; an injected function in tests).
    /// </summary>
    /// <exception cref="InvalidOperationException">
    /// if X402_API_KEY or X402_SECRET is missing.
    /// </exception>
    public static X402Config Resolve(Func<string, string?>? getenv)
    {
        var g = getenv ?? Environment.GetEnvironmentVariable;

        var keyId = NullToEmpty(g("X402_API_KEY"));
        var secret = NullToEmpty(g("X402_SECRET"));
        if (keyId.Length == 0 || secret.Length == 0)
        {
            throw new InvalidOperationException(
                "x402: X402_API_KEY and X402_SECRET must be set (server-side env)");
        }

        var mode = NullToEmpty(g("X402_ENV")).ToLowerInvariant();
        if (mode.Length == 0)
        {
            mode = "production";
        }

        var baseUrl = NullToEmpty(g("X402_BASE_URL"));
        if (baseUrl.Length == 0)
        {
            baseUrl = DefaultBaseUrl;
        }
        baseUrl = baseUrl.TrimEnd('/');

        var env = mode == "sandbox" ? "sandbox" : "production";
        return new X402Config(keyId, secret, baseUrl, env);
    }

    private static string NullToEmpty(string? s) => s ?? string.Empty;
}
